File: MedianOfTwoSortedArrays/medianOfTwoSortedArray.go
package main

import (
	"fmt"
	"math"
)

// https://leetcode.com/problems/median-of-two-sorted-arrays/

type test struct {
	array1 []int
	array2 []int
}


func findMedianSortedArrays(nums1 []int, nums2 []int) (median float64) {
	for _, a:= nums2 {
		mid := math.Round(len(nums1))
		nums1 := append(num1)
		if a > mid{
			if a < nums1[len(nums1)-1]{
				// append between
				copy()
			}else{
				// append after
			}
		}else{
			if a > nums1[0]{
				// append between
			}else{
				// append after
			}
		}
	}
}

func main() {
	testcases := []test{
		test{array1: []int{1,5, 9}, array2: []int{2,4, 7}},
		// test{array1: []int{1, 2}, array2: []int{3, 4}},
		// test{array1: []int{0, 0}, array2: []int{0, 0}},
		// test{array1: []int{}, array2: []int{1}},
		// test{array1: []int{2}, array2: []int{}},
	}
	for _, testcase := range testcases {
		fmt.Println(findMedianSortedArrays(testcase.array1, testcase.array2))
	}
}

File: MedianOfTwoSortedArrays/medianOfTwoSortedArray.py
from typing import List
import math


def findMedianSortedArrays(nums1: List[int], nums2: List[int]) -> float:
    joint_nums = nums1 + nums2
    joint_nums.sort()
    if (len(joint_nums) % 2 == 0):
        return (joint_nums[int(len(joint_nums)/2)-1] + joint_nums[int(len(joint_nums)/2)])/2
    else:
        return joint_nums[math.floor(len(joint_nums)/2)]

#
# def findMedianSortedArrays2(nums1: List[int], nums2: List[int]) -> float:
#     joint_nums = nums1 + nums2
#     joint_nums.sort()
#     if (len(joint_nums) % 2 == 0):
#         return (joint_nums[int(len(joint_nums)/2)-1] + joint_nums[int(len(joint_nums)/2)])/2
#     else:
#         return joint_nums[math.floor(len(joint_nums)/2)]


print(findMedianSortedArrays([1, 2, 4], [3, 5]))
